kmp-library/src/commonTest/kotlin/com/novoda/aws/v4/signer/HttpRequestTest.kt

```diff
@@ -10,28 +10,26 @@
 
   Copyright 2016 the original author or authors.
  */
-package uk.co.lucasweb.aws.v4.signer;
+package com.novoda.aws.v4.signer
 
-import org.junit.Test;
+import io.ktor.http.URLBuilder
+import kotlin.test.Test
+import kotlin.test.assertEquals
 
-import java.net.URI;
+class HttpRequestTest {
 
-import static org.assertj.core.api.Assertions.assertThat;
-
-/**
- * @author Richard Lucas
- */
-public class HttpRequestTest {
+    @Test
+    fun shouldGetHttpMethod() {
+        assertEquals(HttpRequest.create("GET", "/test").method, "GET")
+    }
 
     @Test
-    public void shouldGetHttpMethod() throws Exception {
-        assertThat(new HttpRequest("GET", new URI("http://localhost/test")).getMethod())
-                .isEqualTo("GET");
+    fun shouldGetQueryFromUrl() {
+        assertEquals(HttpRequest.create("GET", URLBuilder("http://localhost/test?test=one&hello=world").build()).query, "test=one&hello=world")
     }
 
     @Test
-    public void shouldGetQuery() throws Exception {
-        assertThat(new HttpRequest("GET", new URI("http://localhost/test?test=one&hello=world")).getQuery())
-                .isEqualTo("test=one&hello=world");
+    fun shouldGetQueryFromPathAndQuery() {
+        assertEquals(HttpRequest.create("GET", "/test?test=one&hello=world").query, "test=one&hello=world")
```

Comment:
I was looking for a test of `http://localhost/?foo=bar` aka param with no path


I don't think it needs testing but emojis in urls are so hot right now :) 

http://factmyth.com/factoids/you-%F0%9F%91%8F-can-%F0%9F%91%8F-put-%F0%9F%91%8F-emjois-%F0%9F%91%8F-in-%F0%9F%91%8F-urls-%F0%9F%91%8F/